### PR TITLE
KAFKA-12824: Remove Deprecated method KStream#branch

### PR DESCRIPTION
**Pull request by frankvicky**
JIRA: [KAFKA-12824](https://issues.apache.org/jira/browse/KAFKA-12824)

The method branch in both Java and Scala KStream class was deprecated in version 2.8:

- `org.apache.kafka.streams.scala.kstream.KStream#branch`
- `org.apache.kafka.streams.kstream.KStream#branch(org.apache.kafka.streams.kstream.Predicate<? super K,? super V>...)`
- `org.apache.kafka.streams.kstream.KStream#branch(org.apache.kafka.streams.kstream.Named, org.apache.kafka.streams.kstream.Predicate<? super K,? super V>...)`

Now it's time to remove them.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by chia7712 — APPROVED** [review]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java [hunk 1]
```diff
@@ -450,22 +450,6 @@ public KStream<K, V> peek(final ForeachAction<? super K, ? super V> action,
...
-    @Override
-    @SuppressWarnings("unchecked")
-    public KStream<K, V>[] branch(final Named named,
-                                  final Predicate<? super K, ? super V>... predicates) {
-        Objects.requireNonNull(named, "named can't be null");
-        return doBranch(new NamedInternal(named), predicates);
-    }
-
     @SuppressWarnings({"unchecked", "rawtypes"})
     private KStream<K, V>[] doBranch(final NamedInternal named,
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:454]: @frankvicky -- this method in unused now -- can you do a follow up PR to remove it (using the same Jira).

Comment (frankvicky) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:454]: Hi @mjsax 
No problem, I will do it

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:454]: Cool. Thx. Please share a link to the PR here so I don't miss it :)

Comment (frankvicky) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:454]: Yeah, sure.
Here is the link:
#16981

**Pull request merged**